Fast and Flexible GPU Quantization for Transformers.md

```diff
@@ -1,82 +1,193 @@
+Dynamic quantization tends to be more accurate but requires additional computational overhead for online scalar calibration. As a result, **we only consider static quantization on GPU** because scalar reduction, relative to an INT8 matmul, can be costly and limit performance gains.
 
 ### Quantization Granularity
-A final distinction to be made is how we share quantization parameters between elements of our parameters and activations. For the rest of this blog, we will use the following diagram to describe a matmul:
+A final distinction to be made is how we quantization parameters are shared between elements of our parameters and activations. Throughout this blog, we'll use the following diagram to illustrate a matmul:
 
 ![](_attachments/Pasted%20image%2020230313162138.png)
 
-The simplest approach is to use the same scale factor for all elements of W (and likewise for X). This is **per-tensor** quantization.
+The simplest approach is to use the same scale factor for all elements of $W$ (and likewise for $X$). This is known as **per-tensor** quantization.
 
-It’s also feasible to share quantization parameters between some subgroups of each input matrix. A common choice is to assign a specific scale factor to each column of W. This is **per-channel (aka per-column) quantization**.
+It’s also feasible to share quantization parameters between some subgroups of each input matrix. A popular options is to assign a specific scale factor to each column of $W$, referred to as **per-channel (or per-column) quantization**.
 
-## Theory
-Having covered the basics of quantization, we’ll now look at the important concepts in their implementation.
+## Important Concepts
+With the fundamentals of quantization covered, let's explore the important concepts in its implementation.
 
 ### Calibration
 
-As explained above, calibration is the process of obtaining activation quantization parameters. We pass a few batches of data through the model to measure the distribution over activations.
+Calibration, as mentioned above, involves obtaining activation quantization parameters by passing several batches of data through the model to measure activation distribution.
 
-There are multiple ways of obtaining a clipping range from these activations. These include:
+There are multiple methods to derive a clipping range from these activations, such as:
 
-* Taking a simple min/max
+* Using a simple min/max
 * Minimising KL Divergence between the input and quantized distributions
 * Minimising the Mean-Squared Error between input and quantized distributions
 
-We found that the final approach was most performant (although this may be different for different models).
+We found that the final approach was most performant (although this may vary for different models).
+
+To perform the calibration process, we recommend using TensorRT’s PyTorch Quantization Toolkit [5]. 
```

Comment:
Not sure whether we should recommend this, it's unmaintained for two years, if we had more time we'd probs want to use PyTorch for calibration

Reply:
Maybe we still say its an option, in addition to PyTorch QDQ nodes

Reply:
Have mentioned PyTorch stuff, and that TRT PT Quantization isn't regularly maintained. Also eluded to PyTorch 2.0, mentioning that their quant stuff is in beta